src/platform/intel/ace/include/ace/version.h

```diff
@@ -8,6 +8,22 @@
 #ifndef __ACE_VERSION_H__
 #define __ACE_VERSION_H__
 
-#define HW_CFG_VERSION		0
+#define ACE_VERSION_1_5 0x10500 /* TGL */
+#define ACE_VERSION_2_0 0x20000 /* MTL */
+#define ACE_VERSION_2_5 0x20500 /* LNL */
+#define ACE_VERSION_3_0 0x30000 /* PTL */
+
+/* ACE version defined by CONFIG_ACE_VER_*/
```

Comment:
I like the idea but this is a pretty significant change so please submit this in a separate PR for... Continuous Integration reasons: we want to test this alone first in daily tests and in developers' workspaces. Kconfig and the pre-processor are error-prone; we had very many issues there in the past.

It will also be much more visible and easier and faster to review and merge separately:
https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#defining-smaller-prs

To avoid git conflicts you can of course you can submit all values in one go, including the "PTL" bits which will only be used later. They won't harm.

Pro tip: you do NOT need to create a new git branch to submit a separate PR. I rarely ever submit HEAD to Github, almost always push HEAD~n for some value of n. I use git branches only when I need to move something out of sight, otherwise I just rotate git commits with interactive git rebase.

Reply:
> I like the idea but this is a pretty significant change so please submit this in a separate PR [...] It will also be much more visible

I mean this generic idea deserves a full review, not being "sneaked" in the middle of a large and mostly unrelated platform enabling PR.

Reply:
It is nothing new, I've just filled version.h according to what was done before in cavs https://github.com/thesofproject/sof/blob/main/src/platform/intel/cavs/include/cavs/version.h

Reply:
It's brand new for SOF with Zephyr, so I think it still deserves a separate review and not being sneaked in a PTL-specific PR. While all the code changes seem Intel-specific, it's modifying generic `/ipc4/` files which are not Intel-specific.

From a testing and CI perspective, even the smallest and simplest changes can break everything when they're not tested, most recent example:
- https://github.com/zephyrproject-rtos/zephyr/pull/72511#discussion_r1638965757

Reply:
After fast internal discussion we can close that one.